### PR TITLE
Add NetworkFileItemClassify

### PR DESCRIPTION
**Pull request by notspiff**
## Description
Continue moving classifiers out of CFileItem

## Motivation and context
Reduce CFileItem API

## How has this been tested?
It builds + I have added tests

## What is the effect on users?
None

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [x] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

**Review by garbear — APPROVED** [review]
Love that there's test cases. It also looks like, now that NETWORK:: is introduced, we've settled on a good pattern of namespace usings.

File: xbmc/network/NetworkFileItemClassify.cpp [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+#include "FileItem.h"
+#include "utils/URIUtils.h"
+
+namespace KODI::NETWORK
+{
+
+bool IsInternetStream(const CFileItem& item, const bool bStrictCheck /* = false */)
+{
+  if (item.HasProperty("IsHTTPDirectory"))
+    return bStrictCheck;
```
Comment (CrystalP) [inline xbmc/network/NetworkFileItemClassify.cpp:20]: Kind of suspicious but it's the existing logic. There isn't any caller with bStrictCheck=true, so the function parameter could be removed.

Comment (notspiff) [reply xbmc/network/NetworkFileItemClassify.cpp:20]: can i do that in a follow-up please?

Comment (CrystalP) [reply xbmc/network/NetworkFileItemClassify.cpp:20]: sure, looks good to me otherwise.

**Pull request merged**